### PR TITLE
8 implement transfer of tile from market to town

### PR DESCRIPTION
**Pull request by Structed**

### CODE REVIEW

**Comment by Structed** [comment]
Currently, there is an issue with the tiles not having the same walls when appearing in town, but only after card refill

**Comment by Structed** [comment]
> Currently, there is an issue with the tiles not having the same walls when appearing in town, but only after card refill

Fixed. Nodes were re-used in market, but walls were never re-set.

**Comment by Structed** [comment]
alcaramba-2022-12-29

File: src/alcaramba/Drawable/Card/TileCardDrawable.gd [hunk 1]
```diff
@@ -7,25 +7,48 @@ var _card_info: TileCard
...
+	$Panel_WallRight.visible = false
+
+func select():
+	set_scale(Vector2(1.5, 1.5))
+	
+func deselect():
+	set_scale(Vector2(1, 1))
+
+# Resets the drawable
+func reset():
```
Comment (zientist) [inline src/alcaramba/Drawable/Card/TileCardDrawable.gd:47]: why is reset needed, when it is the same as deselect

File: src/alcaramba/Drawable/Game/Game.gd [hunk 2]
```diff
@@ -1,8 +1,32 @@
...
+	$Market.get_node("TileMarket").connect("tile_card_selected", self, "_on_tile_card_selected")
+
+
+func _on_tile_card_selected(card_node: TileCardDrawable):
+	# If another tile was clicked, we need to first reset the previous tile
+	if _selected_tile != null && card_node != _selected_tile:
+		_selected_tile.reset()
+
+	_selected_tile = card_node
+	if _town.is_connected("tile_placed", self, "_on_tile_placed"):
```
Comment (zientist) [inline src/alcaramba/Drawable/Game/Game.gd:23]: bitte mir mal live erklären

File: src/alcaramba/Drawable/Card/TileCardDrawable.gd (hunk 1 above)
Comment (Structed) [reply src/alcaramba/Drawable/Card/TileCardDrawable.gd:47]: Not needed, but I felt it's more clear and we might to extend one ore the other later.

If you like, I can remove. If it doesn't bother you, we'll leave it.

File: src/alcaramba/Scripts/Player.gd [hunk 3]
```diff
@@ -6,7 +6,17 @@ var index: int
...
+var town_tiles = {}
 
 func _init(player_index : int, player_name : String):
 	self.index = player_index
 	self.name = player_name
+
+func add_town_tile(tile: TileCard, position: Vector2):
+	var x: int = position.x
+	var y: int = position.y
+	if not town_tiles.has(x):
```
Comment (zientist) [inline src/alcaramba/Scripts/Player.gd:18]: index for dictionary will not work for negative coordinates, possible solution is to set start tile way to high positive coordinates e.g. (100,100)

File: src/alcaramba/Drawable/Market/Market.gd [hunk 4]
```diff
@@ -11,8 +11,8 @@ onready var money_market = get_node("%MoneyMarket")
 
 # Called when the node enters the scene tree for the first time.
 func _ready():
-	Global.stack_tiles.shuffle()
 	money_market.connect("money_card_selected", self, "_on_money_card_selected")
+	_on_EndTurnButton_pressed()
```
Comment (zientist) [inline src/alcaramba/Drawable/Market/Market.gd:15]: kann nur hier kommentieren: endTurn should rescale all tiles to one (if turn is ended with selected tile, it stays big)

**Comment by zientist** [comment]
Approved for merge when previous comment is fixed

File: src/alcaramba/Drawable/Market/Market.gd (hunk 4 above)
Comment (Structed) [reply src/alcaramba/Drawable/Market/Market.gd:15]: Fixed

**Pull request merged**